new/kustomize/overlays/kustomization.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: new-sourcegraph
+resources:
+  - https://github.com/sourcegraph/deploy-sourcegraph/new/resources/sourcegraph?ref=bee/kustomize-new-base
```

Comment:
@beyang Good question, sorry for not clarifying in the PR. I'm currently using the URL with `ref=bee/kustomize-new-base` as an example to demonstrate remote build, since the kustomiziation.yaml in the base layer of any of the `ref=version_number` does not exist yet. 

Once the `kustomiziation.yaml` files for the base folder ([example here](https://github.com/sourcegraph/deploy-sourcegraph/blob/bee/kustomize-new-base/new/resources/sourcegraph/kustomization.yaml)) are merged to main and added to the next release, we can then point it to the new version using the `ref` tag for each release when using remote build. 

File paths should be used for local builds though, I will add these to the docs, or do you think we should just use file paths by default?